packages/admin-ui/src/segment/segment.ts

```diff
@@ -0,0 +1,247 @@
+
+        if (segment.type === 'dayPeriod') {
+          if (key === 'a') {
+            state.setSegment('dayPeriod', 0)
+          } else if (key === 'p') {
+            state.setSegment('dayPeriod', 12)
+          }
+
+          state.next()
+        } else {
+          if (!isNumeric(newValue)) {
+            return
+          }
+
+          const numberValue = parseNumber(newValue)
+          let segmentValue = numberValue
+
+          if (
+            segment.type === 'hour' &&
+            state.dateFormatter.resolvedOptions().hour12 &&
+            numberValue === 12
+          ) {
+            segmentValue = 0
+          } else if (numberValue > (segment.maxValue as number)) {
+            segmentValue = parseNumber(key)
+          }
+
+          state.setSegment(segment.type, segmentValue)
+
+          if (Number(`${numberValue}0`) > (segment.maxValue as number)) {
+            setEnteredKeys('')
+            state.next()
+          } else {
+            setEnteredKeys(newValue)
+          }
+        }
+      },
+      [enteredKeys, next, segment.maxValue, segment.type]
+    )
+
+    const onKeyDown = useCallback(
+      (e: React.KeyboardEvent) => {
+        if (e.ctrlKey || e.metaKey || e.shiftKey || e.altKey) {
+          return
+        }
+
+        switch (e.key) {
+          case 'Enter':
+            e.preventDefault()
+            next()
+            break
+
+          case 'Tab':
+            break
+
+          case 'Backspace': {
+            e.preventDefault()
+            if (isNumeric(segment.text) && !isReadOnly) {
+              const newValue = segment.text.slice(0, -1)
+
+              setSegment(
+                segment.type,
+                newValue.length === 0
+                  ? (segment.minValue as number)
+                  : parseNumber(newValue)
+              )
+              setEnteredKeys(newValue)
+            }
+
+            break
+          }
+
+          default:
+            e.preventDefault()
+            e.stopPropagation()
+            if ((isNumeric(e.key) || /^[ap]$/.test(e.key)) && !isReadOnly) {
+              onNumericKeyDown(e.key)
+            }
+        }
+      },
+      [next, onNumericKeyDown, segment.minValue, segment.text, segment.type]
+    )
+
+    const onFocus = useCallback(() => {
+      setEnteredKeys('')
+    }, [])
+
+    const onMouseDown = useCallback(
+      (e: React.MouseEvent) => e.stopPropagation(),
+      []
+    )
+
+    const elementProps = useMemo(() => {
+      switch (segment.type) {
+        case 'literal':
+          return {
+            state,
+            children: segment.text,
+            tabIndex: disabled ? -1 : 0,
+            disabled,
+            ...htmlProps,
+          }
+
+        case 'era':
+          return {
+            state,
+            children: segment.text,
+            tabIndex: disabled ? -1 : 0,
+            disabled,
+            ...htmlProps,
+          }
+
+        default:
+          return mergeProps(spinButtonProps, {
```

Comment:
suggestion: Maybe it could be create before the switch an object:
```suggestion
          const baseProps = {
            state,
            children: segment.text,
            tabIndex: disabled ? -1 : 0,
            disabled,
            ...htmlProps,
          }
          
          ...
          switch (segment.type) {
             default:
                return mergeProps(spinButtonProps, {
                    ...baseProps,
                    'aria-label': segment.type,
                    onKeyDown: callAllHandlers(htmlOnKeyDown, onKeyDown),
                    onFocus: callAllHandlers(htmlOnFocus, onFocus),
                    onMouseDown: callAllHandlers(htmlOnMouseDown, onMouseDown),
                }
          }
          
          
```

Reply:
Done